src/dom_types.rs

```diff
@@ -1121,6 +1136,33 @@ impl<Ms> El<Ms> {
         }
     }
 
+    /// Converts an element with another message type into an element that can be returned by the
+    /// view function.
+    /// 
+    /// This allows the use of third party components to integrate with your application without
+    /// having to know about your Msg type beforehand. 
+    /// All that is required is that you implement `From<OtherMsg> for Msg` and hand off their
+    /// `OtherMsg` and section of state to their `update()` function (or equivalent) within your `update()` function.
+    ///
+    /// # Note
+    /// There is an overhead to calling this versus keeping all messages under one type.
+    /// The deeper the nested structure of children, the more time this will take to run.
+    pub fn convert_message<OtherMs: Into<Ms>>(other: El<OtherMs>) -> El<Ms> {
```

Comment:
Maybe implement `From<El<OtherMs>> where OtherMs: Into<Ms>`?

Reply:
I think that's what I initially tried, but ran afoul of the case where `OtherMs` is actually `Ms` (because `Ms` implements `Into<Ms>`), and `From<El<Ms>>` can't be reimplemented because the blanket `From<T> for T` impl already covers that case. I'll double-check this and get back to you.

Reply:
Also, perhaps accept a callback to apply to the message, like I did for `Update.map`. This has the advantage that the caller can choose a different transformation, other than `OtherMs::into`. Also, maybe you can store the callback inside the `El` instead of applying it directly, to solve the overhead you mention in the comment.

Reply:
I'll give that a shot.

I am sad to confirm that a From<El<OtherMs>> fails due to a conflicting implementation in core: `impl<T> std::convert::From<T> for T``.

Reply:
I don't think storing the callback with `El` would be helpful, it A: necessitates that the other message type be part of the `El`, which would reduce ergonomics (eg: `El<Msg, OtherMsg>`) and still wouldn't scale beyond one external supported crate, and B: I think that the storage of the mapping function still needs to be done to all children recursively, which would still incur a similar cost.